DQMOffline/L1Trigger/src/L1TTauOffline.cc

```diff
@@ -533,10 +533,20 @@ bool L1TTauOffline::matchHlt(edm::Handle<trigger::TriggerEvent> const& triggerEv
 
   trigger::TriggerObjectCollection trigObjs = triggerEvent->getObjects();
 
-  for (auto trigIndexIt = m_trigIndices.begin(); trigIndexIt != m_trigIndices.end(); ++trigIndexIt) {
-    const vector<string> moduleLabels(m_hltConfig.moduleLabels(*trigIndexIt));
-    const unsigned moduleIndex = m_hltConfig.size((*trigIndexIt)) - 2;
-
+  for (auto& trigIndex : m_trigIndices) {
+    const vector<string> moduleLabels(m_hltConfig.moduleLabels(trigIndex));
+    // V.M. 2023.03.08 Same issue as in the L1TMuonDQMOffline.cc - some modules are behind hltBoolEnd, but we want the last one before the "hltBoolEnd"
+    size_type moduleIndex = 999999;
+    for (size_type ii = moduleLabels.size() - 1; ii > 0; ii--) {
+      if (moduleLabels[ii] == "hltBoolEnd") {
+        moduleIndex = ii - 1;
+        break;
+      }
+    }
+    if (moduleIndex == 999999) {
```

Comment:
Hi @perrotta, I'm getting a compilation error locally due to the change of the definition of ```moduleIndex``` (from ```unsigned int``` to ```size_type```):
```error: unsigned conversion from 'int' to 'trigger::size_type' {aka 'short unsigned int'} changes value from '999999' to '16959' ```

which would make this comparison always false. Should I return to ```unsigned_int```?

Reply:
Yeah, I overlooked that you can't descrive 999999 with just 16 bis.
Unsigned int is fine
My comment was mainly meant to avoid repeated casts when not necessary

Reply:
Thank you! The change has been implemented in the latest commit.